src/gui/qgsattributeform.cpp

```diff
@@ -1824,6 +1826,14 @@ QgsAttributeForm::WidgetInfo QgsAttributeForm::createWidgetFromDef( const QgsAtt
 
         QgsEditorWidgetWrapper *eww = QgsGui::editorWidgetRegistry()->create( widgetSetup.type(), mLayer, fldIdx, widgetSetup.config(), nullptr, this, mContext );
         QgsAttributeFormEditorWidget *formWidget = new QgsAttributeFormEditorWidget( eww, widgetSetup.type(), this );
+
+        QgsRelationReferenceWidget *relRefWidget = dynamic_cast<QgsRelationReferenceWidget *>( eww->widget() );
+        if ( relRefWidget )
+        {
+          //This field widget can be a frame for a reference relation, show or hide its label accordingly
+          relRefWidget->setShowLabel( widgetDef->showLabel() );
```

Comment:
I'm not sure this is a good idea to have the same parameter to configure the visibility of two different things:
- the label in the form left column
- the groupbox label in the relation reference embedded form 

Maybe it would be better to add a specific configuration parameter

Reply:
I agree. We can potentially keep the first commit from this PR which actually fix a bug. But to show/hide the groupbox label which is the title of the QgsRelationReferenceWidget we could add a new check box element in the attribute editor configuration view. That would be a new feature though. Is it okay for you ?

Reply:
> Is it okay for you ?
Yes